src/libs/Network/SequentialQueue.js

```diff
@@ -33,6 +40,7 @@ function process() {
 
     const task = _.reduce(persistedRequests, (previousRequest, request) => previousRequest.then(() => {
         currentRequest = Request.processWithMiddleware(request, true);
+        pendingRequests.push(currentRequest);
```

Comment:
My expectation is that the `.finally()` of `process()` won't get called until these all finish anyway.

Reply:
It doesn't, without waiting on pending requests the finally always fires before.

Reply:
Sorry, I'm not really following. What fires "after" the `finally()`? 

These requests run in sequence and the last one completing should mean we are "done". 

Why can we not use the `finally()` to figure out whether the queue is done running?

Reply:
The `finally()` invokes the callback passed to it when the requests on the SequentialQueue have finished firing, but not responding.

We have a very loose definition of the queue being done running here. The `finally` assumes the queue is done running when it has no work left, i.e. it has fired all of its requests.

Reply:
> Sorry, I'm not really following. What fires "after" the finally()?

The persisted requests

If we use `finally`, [this happens](https://github.com/Expensify/App/pull/12219/files#r1009935060) after the first try, can't really explain why this happens yet

Reply:
> These requests run in sequence and the last one completing should mean we are "done".

Ah, I know why it didn't work, it's because there was only a single request in the queue when I tested, and since the `reduce` was done after that single request then we just resolve immediately in [the base case](https://github.com/Expensify/App/blob/005240398e3a43328a07ce8791825fd415d4459f/src/libs/Network/SequentialQueue.js#L37-L39) and run whatever is in the `finally()`.

Reply:
cc ^ @marcaaron

Reply:
Ok, I read that comment and observing what is happening is a great place to start, but we need to figure out why it's happening.